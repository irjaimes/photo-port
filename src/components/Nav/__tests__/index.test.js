import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "..";

afterEach(cleanup);

describe("Nav component", () => {
  // baseline test
  it("renders", () => {
    render(<Nav />);
  });

  // snapshot test
  it("matches component", () => {
    const { asFragment } = render(<Nav />);
    //asser the value comparison
    expect(asFragment).toMatchSnapshot();
  });
});

// to check if camera emoji renders
describe("emoji is visible", () => {
  it("inserts emoji into the h2", () => {
    // Arrange parts of components that need to be accessed
    const { getByLabelText } = render(<Nav />);
    // Assert
    expect(getByLabelText("camera")).toHaveTextContent("ðŸ“¸");
  });
});

// check link visibility
describe("links are visible", () => {
  it("inserts text into the links", () => {
    // Arrange
    const { getByTestId } = render(<Nav />);
    // Assert
    expect(getByTestId("link")).toHaveTextContent("Oh Snap!");
    expect(getByTestId("about")).toHaveTextContent("About");
  });
});
