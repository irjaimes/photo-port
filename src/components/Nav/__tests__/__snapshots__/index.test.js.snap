// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches component 1`] = `[Function]`;
